subprojects/docs/src/docs/userguide/tasks.xml

```diff
@@ -290,52 +290,673 @@
+                            <th>Description</th>
+                        </tr>
+                    </thead>
+                    <tr>
+                        <td><literal>@Input</literal></td>
+                        <td>Any serializable type</td>
+                        <td>A simple input value</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@InputFile</literal></td>
+                        <td><literal>File</literal>*</td>
+                        <td>A single input file (not directory)</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@InputDirectory</literal></td>
+                        <td><literal>File</literal>*</td>
+                        <td>A single input directory (not file)</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@InputFiles</literal></td>
+                        <td><literal>FileCollection</literal>*</td>
+                        <td>A collection of input files (no directories)</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@OutputFile</literal></td>
+                        <td><literal>FileCollection</literal>*</td>
+                        <td>A single output file (not directory)</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@OutputDirectory</literal></td>
+                        <td><literal>File</literal>*</td>
+                        <td>A single output directory (not file)</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@OutputFiles</literal></td>
+                        <td><literal>FileCollection</literal>*</td>
+                        <td>A collection of output files (no directories)</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@OutputDirectories</literal></td>
+                        <td><literal>FileCollection</literal>*</td>
+                        <td>A collection of output directories (no files)</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@Nested</literal></td>
+                        <td>Any custom type</td>
+                        <td>A custom type that may not implement <literal>Serializable</literal> but does
+                            have at least one field or property marked with one of the annotations in this
+                            table. It could even be another <literal>@Nested</literal>.</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@Console</literal></td>
+                        <td>Any type</td>
+                        <td>Indicates that the property is neither an input nor an output. It simply
+                            affects the console output of the task in some way.</td>
```

Comment:
As in? I don't really know what `@Console` is for other than it is used for inputs that affect the console output. Perhaps I could say "such as increasing or decreasing the verbosity of the task."